chapter04/ex_4_5.pl

```diff
@@ -0,0 +1,25 @@
+#!perl
+use 5.010;
+
+&greet( "Fred" );
+&greet( "Barney" );
+&greet( "Wilma" );
+&greet( "Betty" );
+
+sub greet {
+    state @last;
+
+    my $name = shift @_;
+
+    if( @last > 0 ) {
+        #printf "Hi %-10s! I've seen: @last\n", $name ;
+        #printf "Hi %-10s! I've seen: %3d\n", $name,$_  for @last;
+        
+        printf "Hi %-10s! I've seen: %s\n", $name, join( ',', @last ) ;
+    }
```

Comment:
What happens when we get to season 6 and want to `&greet( "The Great Gazoo" );` ?